File: windows-build-20260602-stack-box-direct-real-bepu-fix2/cooked/code/gameplay/MenuPlatformInfoDefinition.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "MenuPlatformInfoDefinition.hpp"
#include "runtime/native_exceptions.hpp"
#include "MenuPlatformInfoDefinition.hpp"
#include "runtime/native_exceptions.hpp"

int32_t MenuPlatformInfoDefinition::get_LineSpacing()
{
return this->LineSpacing;
}

int32_t MenuPlatformInfoDefinition::get_RightMargin()
{
return this->RightMargin;
}

int32_t MenuPlatformInfoDefinition::get_TopMargin()
{
return this->TopMargin;
}

MenuPlatformInfoDefinition::MenuPlatformInfoDefinition(int32_t topMargin, int32_t rightMargin, int32_t lineSpacing) : LineSpacing(0), RightMargin(0), TopMargin(0)
{
    if (topMargin < 0)
    {
throw ([&]() {
auto __ctor_arg_0000004E = "topMargin";
auto __ctor_arg_0000004F = "Top margin must be zero or greater.";
return new ArgumentOutOfRangeException(__ctor_arg_0000004E, __ctor_arg_0000004F);
})();
    }
    if (rightMargin < 0)
    {
throw ([&]() {
auto __ctor_arg_00000050 = "rightMargin";
auto __ctor_arg_00000051 = "Right margin must be zero or greater.";
return new ArgumentOutOfRangeException(__ctor_arg_00000050, __ctor_arg_00000051);
})();
    }
    if (lineSpacing < 0)
    {
throw ([&]() {
auto __ctor_arg_00000052 = "lineSpacing";
auto __ctor_arg_00000053 = "Line spacing must be zero or greater.";
return new ArgumentOutOfRangeException(__ctor_arg_00000052, __ctor_arg_00000053);
})();
    }
this->TopMargin = topMargin;
this->RightMargin = rightMargin;
this->LineSpacing = lineSpacing;
}

File: assets/codebase/menu.authoring/MenuPlatformInfoDefinition.cs
namespace city.menu {
    /// <summary>
    /// Describes the layout used for the demo-disc platform information overlay in authored city menu scenes.
    /// </summary>
    public sealed class MenuPlatformInfoDefinition {
        /// <summary>
        /// Initializes one platform-info overlay definition.
        /// </summary>
        /// <param name="topMargin">Distance in authored canvas pixels from the top edge.</param>
        /// <param name="rightMargin">Distance in authored canvas pixels from the right edge.</param>
        /// <param name="lineSpacing">Vertical spacing in authored canvas pixels between the two text lines.</param>
        public MenuPlatformInfoDefinition(int topMargin, int rightMargin, int lineSpacing) {
            if (topMargin < 0) {
                throw new ArgumentOutOfRangeException(nameof(topMargin), "Top margin must be zero or greater.");
            }
            if (rightMargin < 0) {
                throw new ArgumentOutOfRangeException(nameof(rightMargin), "Right margin must be zero or greater.");
            }
            if (lineSpacing < 0) {
                throw new ArgumentOutOfRangeException(nameof(lineSpacing), "Line spacing must be zero or greater.");
            }

            TopMargin = topMargin;
            RightMargin = rightMargin;
            LineSpacing = lineSpacing;
        }

        /// <summary>
        /// Gets the distance in authored canvas pixels from the top edge.
        /// </summary>
        public int TopMargin { get; }

        /// <summary>
        /// Gets the distance in authored canvas pixels from the right edge.
        /// </summary>
        public int RightMargin { get; }

        /// <summary>
        /// Gets the vertical spacing in authored canvas pixels between the platform name and version lines.
        /// </summary>
        public int LineSpacing { get; }
    }
}
